### PR TITLE
Export batch results to csv

### PR DESCRIPTION
**Pull request by carolinemodic**
Export batch tally results as a CSV. 

Sample exported file: 
[votingworks-test-batch-results_sample-county_primary-election_2021-08-19_15-22-03.csv](https://github.com/votingworks/vxsuite/files/7018260/votingworks-test-batch-results_sample-county_primary-election_2021-08-19_15-22-03.csv)

### CODE REVIEW

**Review by eventualbuddha — APPROVED** [review]
Some things that could be better, but overall LGTM.

File: apps/election-manager/src/components/ExportBatchTallyResultsButton.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  const isTestMode = castVoteRecordFiles?.fileMode === 'test'
+  const { election } = electionDefinition!
+
+  const defaultFilename = generateBatchResultsDefaultFilename(
+    isTestMode,
+    election
+  )
+  return (
+    <React.Fragment>
+      <Button small onPress={() => setIsSaveModalOpen(true)}>
```
[the remark this answers is not in the document]
Comment (carolinemodic) [reply apps/election-manager/src/components/ExportBatchTallyResultsButton.tsx:24]: I don't really see a point in wrapping in a useCallback for a relatively simple function like this that fits inline, usually personally if I'm just setting one state variable I'll just do it inline like this but if there is a reason not to do that lmk.

[the remark this answers is not in the document]
Comment (eventualbuddha) [reply apps/election-manager/src/components/ExportBatchTallyResultsButton.tsx:24]: I'm inclined to agree, though it is good to be aware of how that can cause issues due to unnecessary re-renders. IME that is an uncommon issue, though, so 🤷

File: apps/election-manager/src/utils/generateBatchTallyResultsCSV.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    const batchTally = filterTalliesByParamsAndBatchId(
+      fullElectionTally,
+      election,
+      batchId,
+      {}
+    )
+    const contestVoteTotals: string[] = []
+    expandEitherNeitherContests(election.contests).forEach((contest) => {
+      const contestTally = batchTally.contestTallies[contest.id]
+      contestVoteTotals.push(contestTally?.metadata.ballots.toString() ?? '0')
```
[the remark this answers is not in the document]
Comment (eventualbuddha) [reply apps/election-manager/src/utils/generateBatchTallyResultsCSV.ts]: Oh shoot I just realized I wasn't keeping the context of this in mind. These are CSV entries. Definitely should not have grouping separators! Sorry 😢

File: apps/election-manager/src/utils/generateBatchTallyResultsCSV.ts [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    resultCSVRows.push([
+      batchId,
+      batchTally.batchLabel,
+      batchTally.scannerIds.join(', '),
+      batchTally.numberOfBallotsCounted,
+      ...contestVoteTotals,
+    ])
+  })
+
+  return resultCSVRows.map((row) => row.join(',')).join('\n')
```
[the remark this answers is not in the document]
Comment (eventualbuddha) [reply apps/election-manager/src/utils/generateBatchTallyResultsCSV.ts]: Thanks for changing this. That's more what I had in mind, and will make it easier if/when we decide to stream to the file.

File: apps/election-manager/src/utils/generateBatchTallyResultsCSV.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (carolinemodic) [reply apps/election-manager/src/utils/generateBatchTallyResultsCSV.ts]: Ah whoops I should have thought of that too lol, will fix

**Pull request merged**